Automattic/wp-calypso
# Security: Fix Issues with 2FA Flow

This branch fixes #6845 - in honor of #hack-day ( now also #trash-day ) pioneer and bug reporter @isaackeyet 👋 

Currently, in the Two-Step Authentication flow, there are some a couple of bugs ( outlined in #6845 ) that present some confusing messaging, and invalid data to users.  One issue that was reported numerous times was the toast messaging persisting while enabling/disabling 2fa.  In this branch I opted to fix this by just adding a `duration` on the `successNotice` call.

The other issue was that after 2fa is enabled, the "Status" messaging in the component still showed it as disabled until the page was refreshed.  This was due to `connect` not having the `{ pure: false }` configuration set.

While working in the files I also opted to update to extend `Component`, and to utilize the `localize` HoC too.

__To Test__
1. /me/security/two-step
2. Disable two-step, then re-enable. The toast "successfully disabled 2fa" should disappear after a few seconds.
3. Add a number, add a secret via a QR code. Verify the generated codes and do something with the backup codes.
4. Once out of the wizard there's now a prompt "successfully enabled 2fa", it too should disappear after a few seconds.
5. Note the 2fA Status in the top box now says "ON" in green.
